source/mir/ndslice/slice.d

```diff
@@ -910,35 +910,37 @@ public:
     +/
     auto lightScope()() scope return @property
     {
-        return Slice!(LightScopeOf!Iterator, N, kind)(_structure, .lightScope(_iterator));
+        return Slice!(LightScopeOf!Iterator, N, kind, Labels)(_structure, .lightScope(_iterator), _labels);
     }
 
     /// ditto
     auto lightScope()() scope const return @property
     {
-        return Slice!(LightConstOf!(LightScopeOf!Iterator), N, kind)(_structure, .lightScope(_iterator));
+        return Slice!(LightConstOf!(LightScopeOf!Iterator), N, kind, Labels)(_structure,
+            .lightScope(_iterator), _labels);
     }
 
     /// ditto
     auto lightScope()() scope immutable return @property
     {
-        return Slice!(LightImmutableOf!(LightScopeOf!Iterator), N, kind)(_structure, .lightScope(_iterator));
+        return Slice!(LightImmutableOf!(LightScopeOf!Iterator), N, kind, Labels)(_structure,
+            .lightScope(_iterator), _labels);
     }
 
     /// Returns: Mutable slice over immutable data.
-    Slice!(LightImmutableOf!Iterator, N, kind) lightImmutable()() scope return immutable @property
+    Slice!(LightImmutableOf!Iterator, N, kind, Labels) lightImmutable()() scope return immutable @property
     {
-        return typeof(return)(_structure, .lightImmutable(_iterator));
+        return typeof(return)(_structure, .lightImmutable(_iterator), _labels);
     }
 
     /// Returns: Mutable slice over const data.
-    Slice!(LightConstOf!Iterator, N, kind) lightConst()() scope return const @property @trusted
+    Slice!(LightConstOf!Iterator, N, kind, Labels) lightConst()() scope return const @property @trusted
     {
-        return typeof(return)(_structure, .lightConst(_iterator));
+        return typeof(return)(_structure, .lightConst(_iterator), _labels);
     }
 
     /// ditto
-    Slice!(LightImmutableOf!Iterator, N, kind) lightConst()() scope return immutable @property
+    Slice!(LightImmutableOf!Iterator, N, kind, Labels) lightConst()() scope return immutable @property
```

Comment:
a unittest is required `light*` function. It can be a single one unittest for all of them.